examples/counter/src/contract.rs

```diff
@@ -8,13 +8,18 @@ mod state;
 use self::state::Counter;
 use async_trait::async_trait;
 use linera_sdk::{
-    base::SessionId, ApplicationCallResult, CalleeContext, Contract, EffectContext,
-    ExecutionResult, OperationContext, SessionCallResult, SimpleStateStorage,
+    base::{SessionId, WithContractAbi},
+    ApplicationCallResult, CalleeContext, Contract, EffectContext, ExecutionResult,
+    OperationContext, SessionCallResult, SimpleStateStorage,
 };
 use thiserror::Error;
 
 linera_sdk::contract!(Counter);
 
+impl WithContractAbi for Counter {
+    type Abi = counter::CounterAbi;
+}
```

Comment:
What's the advantage of using `WithContractAbi` and a separate `CounterAbi` type, compared to just implementing `ContractAbi` for `Counter` directly? (Similar in the other applications.)